spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/hazelcast/HazelcastAutoConfiguration.java

```diff
@@ -45,6 +46,7 @@
 @ConditionalOnClass(HazelcastInstance.class)
 @ConditionalOnMissingBean(HazelcastInstance.class)
 @EnableConfigurationProperties(HazelcastProperties.class)
+@AutoConfigureAfter(HazelcastClientAutoConfiguration.class)
```

Comment:
:smile: If you recall, that was exactly the situation when PR was initially opened (see [your first comment on this PR](https://github.com/spring-projects/spring-boot/pull/7469#issuecomment-262708514)). I've also been undecided between the two.

I'll try to go back to single `HazelcastAutoConfiguration` again with the next PR update.